elisp/defs.bzl

```diff
@@ -408,6 +427,34 @@ To add a load path entry for the current package, specify `.` here.""",
             doc = "List of `elisp_library` dependencies.",
             providers = [EmacsLispInfo],
         ),
+        "enable_package": attr.bool(
+            doc = """Enable generation of package.el package for this library.
+This value is forced to False if testonly is True.""",
```

Comment:
Why? Is there anything special about testonly that requires this?

Reply:
Libraries used only in tests aren't intended to be packaged and from the ones I've seen it's a little annoying to make them so.  The point of the package support is to enable packaging, so it doesn't seem useful to enable it for testonly.

Reply:
Is it then possible to test package builds at all? For example, I'd like the analysis test in tests/private/defs.bzl to contain a test for the new functionality.

Maybe it would be better to have a tri-state Boolean (e.g. "always", "never", "auto").